tests/Unit/Domain/Test_ElementDistribution.cpp

```diff
@@ -3,306 +3,597 @@
+      domain_creator_2d, 20, std::unordered_set<size_t>{4, 20});
+  test_uniform_element_distribution_construction(
+      domain_creator_2d, 100, std::unordered_set<size_t>{0, 1});
+
+  // 3D
+  test_uniform_element_distribution_construction(domain_creator_3d, 1);
+  test_uniform_element_distribution_construction(
+      domain_creator_3d, 73, std::unordered_set<size_t>{5, 8, 9, 75});
+  test_uniform_element_distribution_construction(
+      domain_creator_3d, 500, std::unordered_set<size_t>{100});
+}
+
+// Test weighted element distribution for 1D, 2D, and 3D. For each dimension,
+// three cases are tested: single proc requested, multiple procs requested with
+// procs to ignore, and more procs requested than elements to distribute.
+void test_weighted_element_distribution(
+    const domain::ElementWeight element_weight,
+    const DomainCreator<1>& domain_creator_1d,
+    const DomainCreator<2>& domain_creator_2d,
+    const DomainCreator<3>& domain_creator_3d) {
+  // 1D
+  test_weighted_element_distribution_construction(element_weight,
+                                                  domain_creator_1d, 1);
+  test_weighted_element_distribution_construction(
+      element_weight, domain_creator_1d, 10, std::unordered_set<size_t>{4, 6});
+  test_weighted_element_distribution_construction(
+      element_weight, domain_creator_1d, 33, std::unordered_set<size_t>{7});
+
+  // 2D
+  test_weighted_element_distribution_construction(element_weight,
+                                                  domain_creator_2d, 1);
+  test_weighted_element_distribution_construction(
+      element_weight, domain_creator_2d, 20, std::unordered_set<size_t>{4, 20});
+  test_weighted_element_distribution_construction(
+      element_weight, domain_creator_2d, 100, std::unordered_set<size_t>{0, 1});
+
+  // 3D
+  test_weighted_element_distribution_construction(element_weight,
+                                                  domain_creator_3d, 1);
+  test_weighted_element_distribution_construction(
+      element_weight, domain_creator_3d, 73,
+      std::unordered_set<size_t>{5, 8, 9, 75});
+  test_weighted_element_distribution_construction(
+      element_weight, domain_creator_3d, 500, std::unordered_set<size_t>{100});
+}
 }  // namespace
+
+SPECTRE_TEST_CASE("Unit.Domain.ElementDistribution", "[Domain][Unit]") {
+  // Test cost functions
+  test_uniform_cost_function();
+  test_weighted_cost_function(domain::ElementWeight::NumGridPoints);
+  test_weighted_cost_function(
+      domain::ElementWeight::NumGridPointsAndGridSpacing);
+
+  // Inputs for testing `BlockZCurveProcDistribution`
+
+  // 1D, single block
+  const auto lattice_1d = domain::creators::AlignedLattice<1>(
+      {{{{0.0, 1.0}}}}, {{4}}, {{6}}, {}, {}, {});
+
+  // 2D
+  const auto lattice_2d = domain::creators::AlignedLattice<2>(
+      {{{{0.0, 0.3}}, {{0.0, 0.8, 2.5, 4.9}}}}, {{2, 3}}, {{4, 5}}, {}, {}, {});
+
+  // 3D
+  const auto binary_compact_object_creator =
```

Comment:
Just stop using this domain... It's insanely slow, I would guess that alone will get you the the speedup alone.

Reply:
Both the review comment and this comment are fair points. I used the BBH domain simply because I felt the 1D and 2D cases checked the box of testing simple domains and I thought it was good to include a more complex and realistic domain test case in addition to the simple ones, and I also had it in my head that BBH runs were one of the motivations for this change so I had it on the mind and wanted to make sure it worked. But you're right about the importance of testing the smallest things for unit tests and I also think you're right that the BBH is what is making it so slow, so I'll swap it out. And also a fair point about the distributions. I was thinking of covering Dim and proc cases to make sure the combinations worked, but in some cases they're not actually checking anything new from another one, despite the different case. So, tl;dr you're right, I'll simplify, and thanks for the input!